JSDemos/Demos/Validation/Overview/React/App.js

```diff
@@ -22,247 +22,227 @@ import {
   maskLabel, dateLabel, cityLabel, addressLabel, countryLabel,
 } from './data.js';
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    const currentDate = new Date();
-    this.maxDate = new Date(currentDate.setFullYear(currentDate.getFullYear() - 21));
-    this.validatorInstance = null;
-    this.cityPattern = '^[^0-9]+$';
-    this.namePattern = /^[^0-9]+$/;
-    this.phonePattern = /^[02-9]\d{9}$/;
-    this.phoneRules = {
-      X: /[02-9]/,
-    };
-    this.state = {
-      password: '',
-      confirmPassword: '',
-      passwordMode: 'password',
-      confirmPasswordMode: 'password',
-    };
-    this.passwordButton = {
-      icon: '../../../../images/icons/eye.png',
-      type: 'default',
-      onClick: () => {
-        this.setState({
-          passwordMode: (this.state.passwordMode === 'text' ? 'password' : 'text'),
-        });
-      },
-    };
-    this.confirmPasswordButton = {
-      icon: '../../../../images/icons/eye.png',
-      type: 'default',
-      onClick: () => {
-        this.setState({
-          confirmPasswordMode: (this.state.confirmPasswordMode === 'text' ? 'password' : 'text'),
-        });
+const cityPattern = '^[^0-9]+$';
+const namePattern = /^[^0-9]+$/;
+const phonePattern = /^[02-9]\d{9}$/;
+const phoneRules = {
+  X: /[02-9]/,
+};
+const checkComparison = () => true;
+
+function App() {
+  const currentDate = new Date();
+  const maxDate = new Date(currentDate.setFullYear(currentDate.getFullYear() - 21));
+
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [passwordMode, setPasswordMode] = useState('password');
+  const [confirmPasswordMode, setConfirmPasswordMode] = useState('password');
+
+  const passwordButton = useMemo(() => ({
+    icon: '../../../../images/icons/eye.png',
+    type: 'default',
+    onClick: () => {
+      setPasswordMode(passwordMode === 'text' ? 'password' : 'text');
+    },
+  }), [passwordMode, setPasswordMode]);
+
+  const confirmPasswordButton = useMemo(() => ({
+    icon: '../../../../images/icons/eye.png',
+    type: 'default',
+    onClick: () => {
+      setConfirmPasswordMode(confirmPasswordMode === 'text' ? 'password' : 'text');
+    },
+  }), [confirmPasswordMode, setConfirmPasswordMode]);
+
+  const passwordComparison = useCallback(() => password, [password]);
+
+  const onPasswordChanged = useCallback((e) => {
+    setPassword(e.value);
+    if (confirmPassword) {
+      validatorInstance.validate();
+    }
+  }, [confirmPassword, setPassword, validatorInstance]);
+
+  const onInit = useCallback((e) => {
```

Comment:
Can be moved out of component like as validatorInstance

Reply:
Fixed to using ref for holding ValidatorInstance